probe-rs/src/architecture/arm/core/armv7a.rs

```diff
@@ -948,6 +952,24 @@ impl CoreInterface for Armv7a<'_> {
         }
     }
 
+    fn endianness(&mut self) -> Result<Endian, Error> {
+        Ok(match self.endianness {
```

Comment:
nitpicky, but I'm personally not a fan of `Ok(huge block of code)`, and you also don't need to scope in the None case I think.

Reply:
I reworked this a little. It's a shame that there's no `get_or_try_insert()` function on `Option<T>`...